### ISSUE TITLE
Don't expand short-circuit expressions

### ISSUE THREAD

**Issue by ActuallyACat** [prettier/prettier#4336]
Hi there, 

I use short-circuit / conditional rendering in my components. I understand my JSX is also an expression, hence why everything is being split over multiple lines. This becomes an issue when there are multiple expressions before rendering the JSX. My example is better at explaining this. 

Are there any Prettier options I can use to ignore this behaviour? It becomes very cumbersome, especially in stateless components.

**Prettier 1.12.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKACL2A8ATASwDcA+dbCjHABxIBU4APGJDYGACwIGcA6bmAEN4veMwC+OAPS1ylLHPnBF8jAEIiBOAHcMAMj3rBUCJzgAnOlygBzfXoAUK1VVmYXlHASjUArjGcPLGgAYQ5jGzgAXnYuPnCoPAAbODCIuHFAoKJBJN9o2J5+IRFvPxgGCQwpMncg6TcggEosjEy6ylacACN-GGhW4KgQpIIwAGsYwexNHQxpigB+DE4ihOS4AFFCGHmOoNZV+OMNgGVBIjhp9pdaoOBZ3WWAcm2CGGeMVmfzy+eb1TSXowfpQO6eKSEUhyEAAGhAEGoMAI0G4yFAgnM5gg2gACpiEGiULltIIAJ5o+GCATIABmuW4cHh3XMggmcBgp2obO8NmQMHM+XheAgYDpDKZIG8jPMMFxrJsAFtBOKkoz4QArbiMABCrPZnMEirgABlvHBVeqEf5ygAmS2S7nmGXIEC0pI4uEgajmbwwADqBDwnGQAEZbQAGeE+iCM-2s6iun1wGWXL3mOAAR18BAz8sESpVSHpasljMVBH5grLvJSAEVfKYLcWJfChN1A8GOMhbW3WQRRrYQhBFcrXSYoBb4b5GXRBN0iSXGeJxEA)
```sh
--arrow-parens always
--parser flow
--print-width 120
```

**Input:**
```jsx

      <div>
        {
          !view && !anotherThing &&(
            <p>
              <input
                onChange={this.handleChange}
                value={this.state.inputText} />
            </p>
          )
        }
      </div>
  
```

**Output:**
```jsx
<div>
  {!view &&
    !anotherThing && (
      <p>
        <input onChange={this.handleChange} value={this.state.inputText} />
      </p>
    )}
</div>;

```

**Expected behavior:**
```jsx
<div>
  {!view && !anotherThing && (
    <p>
      <input onChange={this.handleChange} value={this.state.inputText} />
    </p>
  )}
</div>;

```

**Comment by lydell** [comment]
This is something I think about regularly, too. Don't remember why it works they way it does.

**Comment by lukeed** [comment]
Here's an example outside JSX: [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAHANgVwHMBLKNTAXkwG0AdKTTAeiYDp37Hi0B1AQxhgAFpgBkozACc4AR3zFpACgDkkiLgLYAtHiKktk-FGUBKRcE6MrLSwF8T9ALogANCAjYYxdMlB9JagDuAAr+CGjIIHy4gXwAnhFuAEaSfGAA1nAwAMrYaaSEyDCGcG5CMAC2uDxCxPBoeWBw2eF1xABudXGRYGiJIKRocJIwwamEFXzIAGbRQ24AVmgAHgBCqRlZ2XwVcAAypHAzc6UgS8vZBbhwAIr4EPDHuPMgeZJDkpFJfElwuK6vSSkGA8YgAExgQmQAEYAEwABjc2DUQx4qWwkWRcA+7SObjkDzgYw8ERQfDQWigcDgYJpAOkcgURPGkyeLyGFWIRRKbjQV1u90eSGK+FOMB+oIhUKQsLcxT4xFwBQAwhAKqyUNiAKwA-BDAAqP1JItO7VFAEkoLTYNkwEDPABBK3ZGBxa5suC2WxAA)

**Comment by thorn0** [comment]
This issue is specifically about JSX and was actually fixed long ago by #5092.
